test_util/integration_test.py

```diff
@@ -1290,75 +1302,105 @@ def wait_for_endpoint():
 
     r = requests.get(test_cache_url)
 
+    # Handy for diagnosing strange signal service test behavior, not sure if we should
+    # leave these in for master branch, or remove for the final PR.
+    print('TESTING SIGNAL RETURN:\n{}'.format(r.text))
+    print('CACHE SERVER STATUS:\n{}'.format(r.status_code))
+
     r_data = json.loads(r.json())
 
     cluster.destroy_marathon_app(signal_app_definition['id'])
     cluster.destroy_marathon_app(test_server_app_definition['id'])
 
     exp_data = {
-            'Event': 'health',
-            'UserId': '',
-            'ClusterId': '',
-            'Properties': {
-                'provider': 'onprem',
-                'source': 'cluster',
-                'clusterId': '',
-                'customerKey': '',
-                'environmentVersion': '',
-                'variant': 'open'}
-            }
+        'diagnostics': {
+            'event': 'health',
+            'anonymousId': 'test-id',
+            'properties': {}
+        },
+        'cosmos': {
+            'event': 'package_list',
+            'anonymousId': 'test-id',
+            'properties': {}
+        },
+        'mesos': {
+            'event': 'mesos_track',
+            'anonymousId': 'test-id',
+            'properties': {}
+        }
+    }
+
+    # Generic properties which are the same between all tracks
+    generic_properties = {
+        'provider': 'onprem',
+        'source': 'cluster',
+        'clusterId': 'test-id',
+        'customerKey': '',
+        'environmentVersion': '',
+        'variant': 'open'
+    }
+
+    # Insert the generic property data which is the same between all signal tracks
+    for k, v in generic_properties.items():
```

Comment:
Why not:
`exp_data['diagnostics']['properties'] = copy(generic_properties)` for all of these? (The copy may not even be necessary.

If we're updating existing dictionaries, then do a `.update()` - https://docs.python.org/3/library/stdtypes.html#dict.update